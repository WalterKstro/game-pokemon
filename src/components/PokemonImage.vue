<template>
  <div class="container-picture">
      <img v-if="showImagePokemon" 
           draggable="false"  
           state="show"
           class="picture-pokemon picture-origin" 
           :src="generateSourceUrlImage" alt="pokemon">

      <img v-else draggable="false"  
           class="picture-pokemon picture-shadow"
           state="hidden"
           :src="generateSourceUrlImage" alt="pokemon"> 
  </div>
</template>

<script>
export default {
    name: 'PokemonImage',
    props : {
        pokemonId : {
            type : Number,
            required : true
        },
        showImagePokemon : {
            type : Boolean,
            required : true,
            default : false
        }
    },
    computed: {
        generateSourceUrlImage(){
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
        }
    }
}
</script>

<style scope>
.picture-pokemon{
    position: absolute;
    left:50%;
    top:50%;
    transform: translate(-50%,-50%);
    width: 60%;
    height: auto;
    display: block;
}
.picture-shadow{
    filter: brightness(0);
}
.picture-origin{
    filter: brightness(1);
}
.container-picture{
    position: relative;
    max-width: 300px;
    width: 100%;
    height: 300px;
    margin-right: auto;
    margin-left: auto;
}
@media screen and (min-width: 1024px){
    .container-picture{
        margin-right: 0;
        margin-left: 0;
    }
}
 
.show-pokemon{
    filter: brightness(1);
    transition: filter 1s;
}
</style>