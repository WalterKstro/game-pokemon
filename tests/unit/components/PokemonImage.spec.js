import PokemonImage from '@/components/PokemonImage';
import { shallowMount } from '@vue/test-utils';

describe('Test suit of component <PokemonImage/>', () => {
    
    test('Should create a correct snapshot of <PokemonImage/>', () => {

        const wrapper = shallowMount(PokemonImage,{
            propsData: {
                pokemonId: 1,
                showImagePokemon:false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Should show the hidden image and the pokemon one hundred', () => {
        const wrapper = shallowMount(PokemonImage,{
            propsData: {
                pokemonId: 100,
                showImagePokemon:false
            }
        });

        const image = wrapper.find('img');
        const {src, state} = image.attributes();

        expect(src).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg');
        expect(state).toBe('hidden');
    })

    test('Should show the pokemon, if showImagePokemon:true', () => {
        const wrapper = shallowMount(PokemonImage,{
            propsData: {
                pokemonId: 100,
                showImagePokemon:true
            }
        });

        const image = wrapper.find('img');
        const {src, state} = image.attributes();

        expect(src).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg');
        expect(state).toBe('show');
    })
})