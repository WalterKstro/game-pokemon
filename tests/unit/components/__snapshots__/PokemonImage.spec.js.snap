// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit of component <PokemonImage/> Should create a correct snapshot of <PokemonImage/> 1`] = `<div class="container-picture"><img draggable="false" class="picture-pokemon picture-shadow" state="hidden" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon"></div>`;
